CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 3.5.1)
+project(CMaizeSuperbuild VERSION 1.0.0)
+
+include(ExternalProject)
+
+include(GNUInstallDirs)
+
+ExternalProject_Add(
+    "${PROJECT_NAME}_core"
+    SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/src"
+    CMAKE_ARGS -DCMAKE_CXX_COMPILER="${CMAKE_CXX_COMPILER}"
+               -DCMAKE_CXX_STANDARD="${CMAKE_CXX_STANDARD}"
+               -DCMAKE_CXX_EXTENSIONS="${CMAKE_CXX_EXTENSIONS}"
+               -DCMAKE_CXX_STANDARD_REQUIRED="${CMAKE_CXX_STANDARD_REQUIRED}"
```

Comment:
The library should be setting the standard it needs.

Reply:
So is this an unnecessary line of code as well similar to extensions?

Reply:
yes

Reply:
to clarify, passing `CMAKE_CXX_STANDARD` is okay for now as that's something user could have reasonably set, but I wouldn't expect them to pass `CMAKE_CXX_EXTENSIONS` or `CMAKE_CXX_STANDARD_REQUIRED` (nor should they)